toolkit/primitives/domain/src/lib.rs

```diff
@@ -258,6 +256,20 @@ const MAINCHAIN_PRIVATE_KEY_LEN: usize = 32;
 #[byte_string(hex_serialize, hex_deserialize)]
 pub struct MainchainPrivateKey(pub [u8; MAINCHAIN_PRIVATE_KEY_LEN]);
 
+impl MainchainPrivateKey {
+	#[cfg(feature = "std")]
+	pub fn sign(&self, unsigned_message: &[u8]) -> Self {
```

Comment:
Self is MainchainPrivateKey. Return type should be some kind of signature instead of Self, right?